### DISCUSSION TITLE
Not able to complete order with a customer profile?

### DISCUSSION THREAD

**Discussion by matigastirami** [vendurehq/vendure#648] [Q&A] [answered]
Hi DevTeam!

I found a particular problem when developing my site using vendure. I tried to create an order with an admin profile (addItemToOrder with admin bearer token), then I created a plugin to login an customer and assign the activeOrder to this customer, but I'm having the "Not authorized error". Could someone please tell me if this is an error or normal? In case it's normal, can I solve it?

Example of my code trying to resolve that:

```let order = await this.orderService.findOneByCode(ctx, args.orderCode),
            sessionToChange = await this.sessionService.getSessionFromToken(args.token),
            updated = await this.sessionService.setActiveOrder(ctx, sessionToChange, order);
            await this.sessionService.setActiveChannel(sessionToChange, await this.channelService.getDefaultChannel());```

Thanks!

**Comment by michaelbromley** [comment, marked as answer]
Pasting the solution from Slack:

> I could solve the error! The problem was that I didn't assign a customerId to the order, so when Vendure tried to check order.customer.id === user.id the service threw an forbidden error.
